### PR TITLE
Add a parallel full test suite run

### PR DESCRIPTION
**Pull request by steventux**
## Context

This won't be used as a PR merge check and is not relied upon for deployment but this full RSpec run will give us accurate stats for coverage and flakey specs which we can't easily collect via parallel tests.

## Changes proposed in this pull request

Adds a parallel build job to run the entire (RSpec) test suite.

## Guidance to review

This PR build _should_ pass, a simplecov `head-result/.resultset.json` artifact should be created.

See https://github.com/DFE-Digital/apply-for-teacher-training/actions/runs/1517071979 for how this additional job is run in our workflow.

## Link to Trello card

https://trello.com/c/G2D7y5XO/4465-%F0%9F%8F%88-improve-testing-and-coverage

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by cwrw — COMMENTED** [review]
Is there a way to allow Github to trigger a step without contributing to build still appearing as still running? Although everything is green, the build still being yellow. Maybe there is a background or optional step option?

We could do something similar to a deploy step, where this will trigger a full build in another step which does not show a PR as still building, same with a main build?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -203,17 +201,63 @@ jobs:
           INCLUDE_PATTERN: ${{ matrix.include-pattern }}
           EXCLUDE_PATTERN: ${{ matrix.exclude-pattern || ' ' }}
           DEFAULT_FEATURE_FLAG_STATE: ${{ matrix.feature-flags }}
-      - name: Read flakey test results
```
Comment (cwrw) [inline .github/workflows/build.yml:206]: do we want to reinstate this step on the full test suite build?

**Comment by steventux** [comment]
@cwrw :+1: Yes I think it's possible to do this independent of building / testing the PR. https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/.github/workflows/pull_request_checks.yml does this outside of the PR checks, I'll refactor into a different workflow action.

**Comment by steventux** [comment]
image

@cwrw It appears that we can put the test analysis jobs in a separate workflow but they have to be triggered on `push` and `pull_request` so they appear in the list of jobs (incomplete but not required). I can't find a way to background this so that they don't appear in the PR checks, still investigating.

**Comment by steventux** [comment]
Closing for now as we are investigating collating coverage results from test matrix

**Pull request closed without merging**